lighthouse-core/computed/metrics/lantern-metric.js

```diff
@@ -98,15 +99,20 @@ class LanternMetricArtifact {
    * @return {Promise<LH.Artifacts.LanternMetric>}
    */
   static async computeMetricWithGraphs(data, context, extras) {
-    const {trace, devtoolsLog, settings} = data;
+    const {trace, devtoolsLog, gatherContext, settings} = data;
+    if (gatherContext.gatherMode !== 'navigation') {
+      throw new Error(`Lantern metrics can only be computed on navigations`);
```

Comment:
Do you think we can get away without this check (and similar one in `metric.js`)?

We're doing a bunch of work to pass the gather mode in to the metrics, but is it worth it just to avoid some undefined behavior when the metrics are accidentally called during the wrong gather mode during development?

That bing said, you have don't most of the annoying work already so my opinion isn't too strong here.

Reply:
We certainly could (in fact to get tests to pass, I'll need to have a fallback for `gatherContext` for the ads plugin anyway). 

My thinking here is that this direction was all about explicitness. I shared your concerns about whether the explicitness was worth it. The next PR makes `processedNavigation` optional in the metric implementation, and while we still have implicit workarounds like #12595 shows, it didn't feel like the spirit of explicitness to rely on those. This signal is also just critically important to integrity of the metrics, so making it a required input helps make that clear. Extra bonus: while the entire interface of trace passing is breaking anyway, it's a good time to plumb one more thing through for validation.